### PR TITLE
Atan2 converter

### LINKED ISSUE #938: ↔ [Converter] Add support for aten::atan2 in Torch-TensorRT
##
Unsupported operator: aten::atan2(Tensor self, Tensor other) -> (Tensor)

- **Function Schema**:
 aten::atan2(Tensor self, Tensor other) -> (Tensor)

- **Original PyTorch API**:
  torch.atan2(input, other, *, out=None) → [Tensor]

- **Relevant TensorRT Documentation**:
  None
## Additional context

When using Fourier transformation in audio segmentation models, `torch.atan2` will be used. 
Because variable audio input lengths requires dynamic shape, fallback to pytorch would cause runtime error when creating example inputs by `partitioning::generateRandomInputs`. 
It reports 
> RuntimeError: Trying to create tensor with negative dimension -1: [1, 1, -1]

Therefore, we need  convertor for `torch.atan2` to successfully convert such models.

### PR DESCRIPTION
**Pull request by gs-olive**
# Description

This feature adds the atan2 converter, as mentioned in Issue #938.

Fixes #938 

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [ x ] My code follows the style guidelines of this project (You can use the linters)
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ x ] I have made corresponding changes to the documentation
- [ x ] I have added tests to verify my fix or my feature
- [ x ] New and existing unit tests pass locally with my changes
- [ x ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: core/conversion/converters/impl/element_wise.cpp [hunk 1]
```diff
@@ -804,6 +805,93 @@ auto element_wise_registrations TORCHTRT_UNUSED =
...
+               auto atan2 = add_elementwise(
+                   ctx,
+                   nvinfer1::ElementWiseOperation::kSUM,
+                   atan2_intermediate->getOutput(0),
+                   correction_term->getOutput(0),
+                   util::node_info(n) + "_corrected_atan2");
+
+               TORCHTRT_CHECK(atan2, "Unable to create atan2 layer from node: " << *n);
+
+               atan2->setName(util::node_info(n).c_str());
```
Comment (narendasan) [inline core/conversion/converters/impl/element_wise.cpp:890]: I believe that AssociateValueAndITensor should do this for you now

File: core/conversion/converters/impl/element_wise.cpp [hunk 2]
```diff
@@ -804,6 +805,93 @@ auto element_wise_registrations TORCHTRT_UNUSED =
...
+               auto intermediate_div = add_elementwise(
+                   ctx, nvinfer1::ElementWiseOperation::kDIV, self, other, util::node_info(n) + "_intermediate_div");
+               auto atan2_intermediate =
+                   ctx->net->addUnary(*intermediate_div->getOutput(0), nvinfer1::UnaryOperation::kATAN);
+
+               // Constant tensors used for quadrant correction
+               auto ZERO = tensor_to_const(ctx, torch::tensor({0.}));
+               auto ONE = tensor_to_const(ctx, torch::tensor({1.}));
+               auto TWO = tensor_to_const(ctx, torch::tensor({2.}));
+               auto PI = tensor_to_const(ctx, torch::tensor({M_PI}));
```
Comment (narendasan) [inline core/conversion/converters/impl/element_wise.cpp:826]: https://github.com/pytorch/pytorch/blob/087049000b6f884221640f5748346b81e9be9572/c10/util/MathConstants.h

Here is PyTorch's PI definition. I think `c10::pi<float>()` should be right. Please make a note that we'd prefer double from a pytorch perspective.

File: core/conversion/converters/impl/element_wise.cpp [hunk 3]
```diff
@@ -1,3 +1,4 @@
+#include <c10/util/MathConstants.h>
```
Comment (narendasan) [inline core/conversion/converters/impl/element_wise.cpp:1]: small style thing, I know other files might use `<>` to include torch, libraries but use `""` for anything other than  stdlib

Comment (gs-olive) [reply core/conversion/converters/impl/element_wise.cpp]: I updated the `<>` notation to `""` on both this new import, as well as the existing torch/torch.h import in this file

**Pull request merged**